tests/test_mem_eff_attention.py

```diff
@@ -80,6 +80,8 @@ def generate_test_shapes_B_Mq_Mkv_H_K_Kv(op):
     xformers.ops.MemoryEfficientAttentionCutlassOp,
     xformers.ops.MemoryEfficientAttentionFlashAttentionOp,
     xformers.ops.MemoryEfficientAttentionCutlassFwdFlashBwOp,
+    xformers.ops.TritonFlashAttentionOp,
```

Comment:
For the future, it might be worth adding a test for checking about race conditions. [This has been illustrated in this comment in flashattention](https://github.com/HazyResearch/flash-attention/blob/0bf5e50038ee341ece03bfd0c8ff45a6c57aed5a/flash_attn/flash_attn_triton.py#L24), and I know that @danthe3rd has had issues with race conditions in the past so might be good to extend our tests to cover this case

Reply:
We check that a bit by running with very large batches, to ensure the GPU is saturated (so missing `__syncthreads()` cause wrong results). We could also have something similar to what flash is doing tho